game/game/gameMap.lumber

```diff
@@ -28,6 +28,9 @@ clearing(game { clearings: Clearings, .. }, Id, Suit, Slots) :-
 :- pub(forest/2).
 forest(game { forests: Forests }, Id) :- in(forest { position: Id }, Forests).
 
+:- pub(ruin/2).
+ruin(game { ruin_items: RuinItems }, Id) :- in(ruin_item { clearing: Id }, RuinItems).
```

Comment:
Pretty close, need to also check that the item is not owned. Ruin items holds whether or not an item is a ruin item, but does not say anything about where that item is at the moment. If it is not owned, it will be under the ruin, otherwise it will be in the player's owned items box.

The other thing to watch out for is that `game { ruin_items: RuinItems }` is not a complete game, so it will not match. When you don't need the full record you have to include the `..` to explicitly ignore the rest, like `game { ruin_items: RuinItems, .. }`.

Lastly, I think including the item's id in this function will make it more useful later on. Might as well add it in now

```suggestion
:- pub(itemInRuin/3).
itemInRuin(game { ruin_items: RuinItems, owned_items: OwnedItems, .. }, Clearing, Item) :- 
  in(ruin_item { clearing: Clearing, item: Item }, RuinItems),
  notin(owned_item { item: Item, .. }, OwnedItems).
```